### PR TITLE
Add remote groupId, artifactId completion

### PR DESCRIPTION
**Pull request by AObuchow**
Signed-off-by: Andrew Obuchowicz <aobuchow@redhat.com>

### CODE REVIEW

**Comment by AObuchow** [comment]
Work in progress

File: org.eclipse.lsp4xml.extensions.maven/src/main/java/org/eclipse/lsp4xml/extensions/maven/searcher/RemoteRepositorySearcher.java [hunk 1]
```diff
@@ -131,7 +138,82 @@ public static ArtifactVersionSearcher getInstance() {
...
+				String noResults = "No artifacts found.";
+				artifactIds.add(noResults);
+			}
+			return artifactIds;
+		});
+	}
+
+	public CompletableFuture<Set<String>> getGroupIds(Artifact artifactToSearch) {
+		if (artifactToSearch.getGroupId().equals("MissingGroupID")) {
+			// Don't do a remote groupId search if the user hasn't inputed anything
```
Comment (AObuchow) [inline org.eclipse.lsp4xml.extensions.maven/src/main/java/org/eclipse/lsp4xml/extensions/maven/searcher/RemoteRepositorySearcher.java]: Return an empty future/set, not null (could cause NPE in the future).

File: org.eclipse.lsp4xml.extensions.maven/src/main/java/org/eclipse/lsp4xml/extensions/maven/searcher/RemoteRepositorySearcher.java [hunk 2]
```diff
@@ -131,7 +138,82 @@ public static ArtifactVersionSearcher getInstance() {
...
+						artifactToSearch.getArtifactId(), artifactToSearch.getVersion()));
+				e.printStackTrace();
+			}
+			List<ArtifactInfo> artifactInfos = new ArrayList<>();
+			response.getResults().forEach(artifactInfos::add);
+			return artifactInfos;
+		});
+
+		return artifactInfoResults.thenApply(artifactInfos -> {
+			// We don't clear the previous results to save computation
```
Comment (AObuchow) [inline org.eclipse.lsp4xml.extensions.maven/src/main/java/org/eclipse/lsp4xml/extensions/maven/searcher/RemoteRepositorySearcher.java]: Remove comment

File: org.eclipse.lsp4xml.extensions.maven/src/main/java/org/eclipse/lsp4xml/extensions/maven/searcher/RemoteRepositoryIndexSearcher.java [hunk 3]
```diff
@@ -281,4 +291,65 @@ private void initializeContext()
...
+			}
+			List<ArtifactInfo> artifactInfos = new ArrayList<>();
+			if (response != null) {
+				response.getResults().forEach(artifactInfos::add);
+			}
+			return artifactInfos;
+		});
+	}
+
+	private ArtifactInfoFilter createVersionFilter(String versionToCompare, int comparisonType) {
```
Comment (AObuchow) [inline org.eclipse.lsp4xml.extensions.maven/src/main/java/org/eclipse/lsp4xml/extensions/maven/searcher/RemoteRepositoryIndexSearcher.java]: This function was simply moved, it's not new.

File: org.eclipse.lsp4xml.extensions.maven/src/main/java/org/eclipse/lsp4xml/extensions/maven/searcher/RemoteRepositoryIndexSearcher.java [hunk 4]
```diff
@@ -281,4 +291,65 @@ private void initializeContext()
...
+				System.out.println("Warning - could not close context: " + context.getId());
+				e.printStackTrace();
+			}
+		}
+
+	}
+
+//////////////////////////////////// Helper Methods ///////////////////////////////////////////////
+
+	private CompletableFuture<List<ArtifactInfo>> createIndexerQuery(Artifact artifactToSearch,
```
Comment (AObuchow) [inline org.eclipse.lsp4xml.extensions.maven/src/main/java/org/eclipse/lsp4xml/extensions/maven/searcher/RemoteRepositoryIndexSearcher.java]: This function was factored from getArtifactVersions(), the logic itself is not new

**Comment by AObuchow** [comment]
@mickaelistria I think this patch is looking good enough for now. If you have time to look at it, run the test suite locally & let me know if there's anything you think should be changed.

**Comment by mickaelistria** [comment]
I won't be much available to review ot before Friday,feel free to merge.

**Comment by AObuchow** [comment]
> I won't be much available to review ot before Friday,feel free to merge.

Ok thanks for letting me know.

**Pull request merged**